float32 circle_x 
float32 circle_y